src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -571,39 +570,48 @@ def __init__(
             "Couldn't load sampling frequency. Please provide it with the " "'sampling_frequency' argument"
         )
 
+        units_table = self._nwbfile.units
+
+        name_to_column_data = {c.name: c for c in units_table.columns}
+        spike_times_data = name_to_column_data.pop("spike_times").data
+        spike_times_index_data = name_to_column_data.pop("spike_times_index").data
+
+        units_ids = name_to_column_data.pop("unit_name", None)
+        if units_ids is None:
+            units_ids = units_table["id"].data
+
         BaseSorting.__init__(self, sampling_frequency=sampling_frequency, unit_ids=units_ids)
+
         sorting_segment = NwbSortingSegment(
-            nwbfile=self._nwbfile, sampling_frequency=sampling_frequency, timestamps=timestamps
+            spike_times_data=spike_times_data,
+            spike_times_index_data=spike_times_index_data,
+            sampling_frequency=sampling_frequency,
         )
         self.add_sorting_segment(sorting_segment)
 
-        # Add properties:
-        properties = dict()
-        import warnings
-
-        for column in list(self._nwbfile.units.colnames):
-            if column == "spike_times":
-                continue
-
-            # Note that this has a different behavior than self._nwbfile.units[column].data
-            property_values = self._nwbfile.units[column][:]
+        # Add properties
+        properties_to_add = [name for name in name_to_column_data if "index" not in name]
```

Comment:
check for ending with underscore index to be more precise as a filter.

Reply:
@alejoe91 this is done, I also added logic to exclude double nested arrays. This could be supported if we want but should be another PR.